### PR TITLE
Webidl legacy platform object tests

### PR DESCRIPTION
**Pull request by tobie**
This is a work in progress based on https://github.com/heycam/webidl/pull/402 and [TimothyGu/webidl-keys-test](https://github.com/TimothyGu/webidl-keys-test).

### CODE REVIEW

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/262396182)

# Firefox (nightly) #
Testing web-platform-tests at revision 164c47102d73a3dcdf5feb4ca773f442a1c0740a
Using binary /home/travis/build/w3c/web-platform-tests/_venv/firefox/firefox
Starting tests
All results were stable
## All results ##
<details>
<summary>3 tests ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/6786/WebIDL/ecmascript-binding/legacy-platform-object/DefineOwnProperty.html">/WebIDL/ecmascript-binding/legacy-platform-object/DefineOwnProperty.html</a></summary>


|                                Subtest                                | Results |                                                  Messages                                                  |
|-----------------------------------------------------------------------|---------|------------------------------------------------------------------------------------------------------------|
|                                                                       | OK      |                                                                                                            |
| `Test [[DefineOwnProperty]] with no indexed property setter support.` | PASS    |                                                                                                            |
| `Test [[DefineOwnProperty]] with indexed property setter support.`    | FAIL    | `assert_throws: function "() =>\n      Object.defineProperty(select, "0", {get: () => {}})" did not throw` |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/6786/WebIDL/ecmascript-binding/legacy-platform-object/GetOwnProperty.html">/WebIDL/ecmascript-binding/legacy-platform-object/GetOwnProperty.html</a></summary>


|                            Subtest                             | Results | Messages |
|----------------------------------------------------------------|---------|----------|
|                                                                | OK      |          |
| `[[GetOwnProperty]] with getters and no setters`               | PASS    |          |
| `[[GetOwnProperty]] with named property getters and setters`   | PASS    |          |
| `[[GetOwnProperty]] with indexed property getters and setters` | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/6786/WebIDL/ecmascript-binding/legacy-platform-object/Set.html">/WebIDL/ecmascript-binding/legacy-platform-object/Set.html</a></summary>


|                                                                    Subtest                                                                     | Results | Messages |
|------------------------------------------------------------------------------------------------------------------------------------------------|---------|----------|
|                                                                                                                                                | OK      |          |
| `must be a no-op when setting a numeric property on an object which implements an indexed property getter but not a setter in non strict mode` | PASS    |          |
| `must throw when setting a property on an object which implements a indexed property getter but not a setter in strict mode`                   | PASS    |          |
| `must allow setting a property on an object which implements a named property getter but not a setter in non strict mode`                      | PASS    |          |
| `setting a property on an object which implements a named property getter but not a setter must throw in strict mode`                          | PASS    |          |

</details>
</details>

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/262396183)

# Sauce (safari) #
Testing web-platform-tests at revision 164c47102d73a3dcdf5feb4ca773f442a1c0740a
Using binary None
Starting tests
All results were stable
## All results ##
<details>
<summary>3 tests ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/6786/WebIDL/ecmascript-binding/legacy-platform-object/DefineOwnProperty.html">/WebIDL/ecmascript-binding/legacy-platform-object/DefineOwnProperty.html</a></summary>


|                                Subtest                                | Results |                                                  Messages                                                  |
|-----------------------------------------------------------------------|---------|------------------------------------------------------------------------------------------------------------|
|                                                                       | OK      |                                                                                                            |
| `Test [[DefineOwnProperty]] with no indexed property setter support.` | FAIL    | `assert_equals: (assert_prop_desc_equals: property 'configurable') expected true but got false`            |
| `Test [[DefineOwnProperty]] with indexed property setter support.`    | FAIL    | `assert_throws: function "() =>\n      Object.defineProperty(select, "0", {get: () => {}})" did not throw` |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/6786/WebIDL/ecmascript-binding/legacy-platform-object/GetOwnProperty.html">/WebIDL/ecmascript-binding/legacy-platform-object/GetOwnProperty.html</a></summary>


|                            Subtest                             | Results |                                             Messages                                            |
|----------------------------------------------------------------|---------|-------------------------------------------------------------------------------------------------|
|                                                                | OK      |                                                                                                 |
| `[[GetOwnProperty]] with getters and no setters`               | FAIL    | `assert_equals: (assert_prop_desc_equals: property 'configurable') expected true but got false` |
| `[[GetOwnProperty]] with named property getters and setters`   | FAIL    | `assert_equals: (assert_prop_desc_equals: property 'writable') expected true but got false`     |
| `[[GetOwnProperty]] with indexed property getters and setters` | FAIL    | `assert_equals: (assert_prop_desc_equals: property 'configurable') expected true but got false` |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/6786/WebIDL/ecmascript-binding/legacy-platform-object/Set.html">/WebIDL/ecmascript-binding/legacy-platform-object/Set.html</a></summary>


|                                                                    Subtest                                                                     | Results |                                          Messages                                          |
|------------------------------------------------------------------------------------------------------------------------------------------------|---------|--------------------------------------------------------------------------------------------|
|                                                                                                                                                | OK      |                                                                                            |
| `must be a no-op when setting a numeric property on an object which implements an indexed property getter but not a setter in non strict mode` | PASS    |                                                                                            |
| `must throw when setting a property on an object which implements a indexed property getter but not a setter in strict mode`                   | FAIL    | `assert_throws: function "function () { element.childNodes["5"] = "foo"; }" did not throw` |
| `must allow setting a property on an object which implements a named property getter but not a setter in non strict mode`                      | PASS    |                                                                                            |
| `setting a property on an object which implements a named property getter but not a setter must throw in strict mode`                          | PASS    |                                                                                            |

</details>
</details>

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/262396190)

# Chrome (unstable) #
Testing web-platform-tests at revision 164c47102d73a3dcdf5feb4ca773f442a1c0740a
Using binary None
Starting tests

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/262396191)

# Sauce (MicrosoftEdge) #
Testing web-platform-tests at revision 164c47102d73a3dcdf5feb4ca773f442a1c0740a
Using binary None
Starting tests

File: WebIDL/ecmascript-binding/legacy-platform-object/OwnPropertyKeys.html [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+
+    test(function() {
+        var element = document.createElement("div");
+        element.attributes.foo = "some value";
+        element.attributes.bar = "and another";
+        element.setAttribute("id", "foo");
+        element.setAttribute("class", "bar");
+        custom_assert_array_equals(Reflect.ownKeys(element.attributes), ["0", "1", "id", "class", "foo", "bar"]);
+    }, "must enumerate own properties after indexed and named properties even when they're added first");
```
Comment (TimothyGu) [inline WebIDL/ecmascript-binding/legacy-platform-object/OwnPropertyKeys.html]: I believe this is missing the `2 ** 40` case I had in my repo, which would be ordered in the front of named own properties for ordinary objects but chronologically for legacy platform objects.

Comment (tobie) [inline WebIDL/ecmascript-binding/legacy-platform-object/OwnPropertyKeys.html]: Yes. I wanted to follow the path it takes in the various algorithms before adding it.

**Review by bzbarsky — COMMENTED** [review]
So we should add the rest of the bits from https://github.com/TimothyGu/webidl-keys-test/blob/master/test-storage.js but we should resolve the symbol issue at https://github.com/heycam/webidl/issues/175 first because without that named setter behavior in the face of symbols is kinda undefined.

**Comment by w3c-bots** [comment]
# Build PENDING

Started: None
Finished: None





View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/18177)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/271931764)

**Comment by foolip** [comment]
This PR keeps coming up when I look for the oldest PRs with no action. @TimothyGu, you self-assigned this, do you still intend to push it along? (If nobody wants to work on it anymore we can just as well close it.)

**Comment by TimothyGu** [comment]
Unfortunately, I don't think I have the bandwidth to push this forward :disappointed:

**Comment by bzbarsky** [comment]
@TimothyGu Is there a summary of what's left to do here?

**Review by bzbarsky — COMMENTED** [review]
Sorry for the lag here... I just reviewed this, and apart from the assert_throws bits that need to get updated to, this looks good.

I'll see if I can get the merge done...

**Review by bzbarsky — APPROVED** [review]

**Pull request merged**